modules/dataframe/src/properties/columnmetadatalistproperty.cpp

```diff
@@ -0,0 +1,145 @@
+    : ListProperty(std::string(identifier), std::string(displayName),
+                   std::make_unique<ColumnMetaDataProperty>("column1", "Column 1"), 0,
+                   ListPropertyUIFlag::Static, invalidationLevel, semantics) {
+
+    setPort(port);
+}
+
+ColumnMetaDataListProperty::ColumnMetaDataListProperty(const ColumnMetaDataListProperty& rhs)
+    : ListProperty(rhs) {}
+
+ColumnMetaDataListProperty* ColumnMetaDataListProperty::clone() const {
+    return new ColumnMetaDataListProperty(*this);
+}
+
+void ColumnMetaDataListProperty::setPort(DataFrameInport& inport) {
+    inport_ = &inport;
+
+    onChangeCallback_ = inport_->onChangeScoped([&]() {
+        if (inport_->hasData()) {
+            updateColumnProperties(*inport_->getData());
+        }
+    });
+    if (inport_->hasData()) {
+        updateColumnProperties(*inport_->getData());
+    }
+}
+
+MetaDataOwner ColumnMetaDataListProperty::getColumnMetaData(size_t columnIndex) const {
+    MetaDataOwner metadata;
+
+    auto p = util::find_if_or_null(getProperties(), [columnIndex](const Property* p) {
+        return static_cast<const ColumnMetaDataProperty*>(p)->getColumnIndex() == columnIndex;
+    });
+    if (p) {
+        auto colprop = static_cast<const ColumnMetaDataProperty*>(p);
+        metadata.setMetaData<DoubleVec2MetaData>("DataRange", colprop->getRange());
+    }
+
+    return metadata;
+}
+
+ColumnMetaDataListProperty& ColumnMetaDataListProperty::resetToDefaultState() {
+    NetworkLock lock(this);
+    clear();
+    if (inport_ && inport_->hasData()) {
+        updateColumnProperties(*inport_->getData());
+    }
+    return *this;
+}
+
+void ColumnMetaDataListProperty::updateColumnProperties(const DataFrame& dataframe) {
+    if (dataframe.getNumberOfColumns() <= 1) return;
+
+    auto columnRange = [](auto& col) {
+        return col->getBuffer()
+            ->getRepresentation<BufferRAM>()
+            ->dispatch<dvec2, dispatching::filter::Scalars>([&](auto br) {
+                auto [min, max] = util::bufferMinMax(br, IgnoreSpecialValues::No);
+                return dvec2{min.x, max.x};
+            });
+    };
+
+    auto toRemove = util::transform(
+        getProperties(), [](Property* p) { return static_cast<ColumnMetaDataProperty*>(p); });
+
+    NetworkLock lock(this);
+    for (const auto&& [idx, col] : util::enumerate<int>(dataframe)) {
+        auto it = util::find_if(toRemove, [header = col->getHeader()](auto p) {
+            return p->getDisplayName() == header;
+        });
+        if (it != toRemove.end()) {
+            (*it)->setColumnIndex(idx);
+            (*it)->setRange(columnRange(col));
+            util::erase_remove(toRemove, *it);
+        } else {
+            auto p = static_cast<ColumnMetaDataProperty*>(constructProperty(0));
+            p->setDisplayName(col->getHeader());
+            p->setColumnIndex(idx);
+            p->setRange(columnRange(col));
+        }
```

Comment:
You need to handle setting the default state here to I think. Otherwise we will deserialize an old range if the input data changes, which could be bad...